### PR TITLE
repair/repair.cc: do_repair_ranges(): prevent stalls when skipping ranges

### PR DESCRIPTION
**Pull request by denesb**
We have observed do_repair_ranges() receiving tens of thousands of ranges to repairs on occasion. do_repair_ranges() repairs all ranges in parallel, with parallel_for_each(). This is normally fine, as the lambda inside parallel_for_each() takes a semaphore and this will result in limited concurrency.
However, in some instances, it is possible that most of these ranges are skipped. In this case the lambda will become synchronous, only logging a message. This can cause stalls beacuse there are no opportunities to yield. Solve this by adding an explicit yield to prevent this.

Fixes: #14330

### CODE REVIEW

**Review by michoecho — APPROVED** [review]

**Comment by nyh** [comment]
I see CI has already passed, by @denesb pushed another version so I'll wait for another CI pass to merge this.
@avikivity there are rumors you decided on another merge freeze, but I didn't get the memo (did you send one?) so until I do, I'll continue to merge PRs that successfully pass CI. If CI never passes, we'll get a merge freeze automatically ;-)

**Comment by denesb** [comment]
> I see CI has already passed, by @denesb pushed another version so I'll wait for another CI pass to merge this. @avikivity there are rumors you decided on another merge freeze, but I didn't get the memo (did you send one?) so until I do, I'll continue to merge PRs that successfully pass CI. If CI never passes, we'll get a merge freeze automatically ;-)

I pushed a typo fix, but I'm fine with waiting for another CI.

The unofficial merge freeze was lifted yesterday evening after we got some successful next jobs passing.

**Comment by asias** [comment]
Looks good to me. Thanks!

**Pull request closed without merging**
